compose.py

```diff
@@ -0,0 +1,305 @@
+# Top-level include/core not copied so --chown does not set it correctly,
+# so explicit set on all of include
+RUN chown -R triton-server:triton-server include
+'''
+    with open(os.path.join(ddir, dockerfile_name), "w") as dfile:
+        dfile.write(df)
+
+
+### add additional backends needed
+def add_requested_backends(ddir, dockerfile_name, backends):
+    df = "# Copying over backends \n"
+    for backend in backends:
+        if (backend.lower()
+                in (EXAMPLE_BACKENDS + CORE_BACKENDS + NONCORE_BACKENDS)):
+            df += '''COPY --chown=1000:1000 --from=full /opt/tritonserver/backends/{} /opt/tritonserver/backends/{}    
+'''.format(backend, backend)
+        else:
+            log("Cannot create container from unsupported backend: " + backend)
+
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+def add_requested_repoagents(ddir, dockerfile_name, repoagents):
+    df = "#  Copying over repoagents \n"
+    for ra in repoagents:
+        if (ra in EXAMPLE_REPOAGENTS):
+            df += '''COPY --chown=1000:1000 --from=full /opt/tritonserver/repoagents/{} /opt/tritonserver/repoagents/{}    
+'''.format(ra, ra)
+        else:
+            log("Cannot create container from unsupported repoagent: " + ra)
+
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+def append_workdir(ddir, dockerfile_name, workdir_path):
+    df = '''WORKDIR {}'''.format(workdir_path)
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+def install_dependencies(ddir, dockerfile_name):
+# TODO: remove libnvidia-ml-dev after 21.06 is launched
+    df = '''
+# Ensure apt-get won't prompt for selecting options
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+            libnvidia-ml-dev \
+            libre2-dev \
+            libb64-dev \
+            pciutils && \
+    rm -rf /var/lib/apt/lists/*
+'''
+
+# Add dependencies needed for python backend
+    if 'python' in FLAGS.backend:
+        df += '''
+# python3, python3-pip and some pip installs required for the python backend
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+         python3 \
+         python3-pip && \
+    pip3 install --upgrade pip && \
+    pip3 install --upgrade wheel setuptools && \
+    pip3 install --upgrade grpcio-tools grpcio-channelz numpy && \
+    rm -rf /var/lib/apt/lists/*
+'''
+    df += '''
+# Add entrypoint script
+WORKDIR /opt/tritonserver
+COPY --chown=1000:1000 --from=full /opt/tritonserver/nvidia_entrypoint.sh .
+ENTRYPOINT ["/opt/tritonserver/nvidia_entrypoint.sh"]
+'''
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+### create build container
+def build_docker_container(ddir, dockerfile_name, container_name):
+    # Before attempting to run the new image, make sure any
```

Comment:
You shouldn't have to remove existing image before building.

Reply:
This logic was copied over from `container_build()` though. A bit confused why we don't need to do it here?

Reply:
I believe the code you copied was for the case where we wanted to create an image by saving a container... and in that case you need to delete any existing image. In this case you are just running "docker build" and that will overwrite an existing image so no need to delete it first.